gtest_parallel.py

```diff
@@ -358,6 +364,56 @@ def dump_to_file_and_close(self):
 
 # Record of test runtimes. Has built-in locking.
 class TestTimes(object):
+  class LockedFile(object):
+    def __init__(self, filename, mode):
+      self._filename = filename
+      self._mode = mode
+      self._fo = None
+
+    def __enter__(self):
+      self._fo = open(self._filename, self._mode)
+
+      try:
+        if sys.platform == 'win32':
+          # We make sure that we lock (and unlock below) always
+          # the same region in file.
+          position = self._fo.tell()
+          self._fo.seek(0)
+
+          # We are locking here fixed location in file to use it as
+          # an exclusive lock on entire file.
+          msvcrt.locking(self._fo.fileno(), msvcrt.LK_LOCK, 1)
+
+          self._fo.seek(position)
```

Comment:
Do we need to support opening at end of file? Otherwise we can just always open at 0?